src/index.ts

```diff
@@ -37,10 +41,157 @@ function gistIdFromTwiddleUrl(twiddleUrl: string): string {
+
+    // tslint:disable-next-line: no-bitwise
+    newZip.addFile(newFileName, (fileContent as unknown) as Buffer, '', 0o644 << 16);
+  });
+
+  // Generate the project root directory.
+  generateDir(targetDir);
+  newZip.extractAllTo(targetDir);
+  // tslint:disable-next-line: no-console
+  console.log(`Files extracted to path: ${targetDir}`);
+  return targetDir;
+}
+
+/**
+ * Returns the json object for a given path
+ * @param targetDir
+ * @param fileName
+ */
+function getJsonFileData(targetDir: string, fileName: string) {
+  const file = `${targetDir}/${fileName}.json`;
+  const twiddleData = JSON.parse(fs.readFileSync(file, 'utf8'));
+  return twiddleData;
+}
+
+/**
+ * Normalize the generated twiddle to a ember-like structure.
+ * @param targetDir
+ */
+function normalizeTwiddle(targetDir: string) {
+  const appDir = `${targetDir}/app`;
+  generateDir(appDir);
+  const files = fs.readdirSync(targetDir);
+  for (let i in files) {
+    const file = files[i];
+    const dirPath = file.split('.');
+    const fileName = dirPath.splice(-2, 2);
+    const newPath = `${appDir}/${dirPath.join('/')}`;
+    const filePath = `${newPath}/${fileName.join('.')}`;
+    // If the file parts length is more than 2 then there is a directory structure expected.
+    if (file.split('.').length > 2) {
+      generateDir(`${appDir}/${dirPath.join('/')}`);
+    }
+    if (fileName.length === 2 && fs.existsSync(newPath)) {
+      try {
+        if (fs.existsSync(filePath)) {
+          continue;
+        } else {
+          fs.renameSync(`${targetDir}/${file}`, filePath);
+        }
+      } catch (err) {
+        throw err;
+      }
+    }
+  }
+}
+
+/**
+ * Augment the already generated ember-like twiddle project app into a real ember app
+ * by augmenting the real Ember app's structure on top of the twiddle app.
+ * @param twiddleData
+ * @param projectName
+ */
+async function augmentEmberApp(twiddleData: any, projectName: string) {
+  const twiddleItem = JSON.parse(JSON.stringify(twiddleData));
+  const emberVersion = twiddleItem.dependencies.ember.split('.');
+  emberVersion.pop();
+  const emberAppDir = await unzipAndExtractFromUrl(
+    `https://${GIT_DOMAIN}/ember-cli/ember-new-output`,
+    `archive/v${emberVersion.join('.')}.0.zip`,
+    `./${projectName}`
+  );
+  if (emberAppDir) {
+    const packageData = getJsonFileData(emberAppDir, 'package');
+    const packageDevDep = packageData.devDependencies;
+    const twiddleAddons = twiddleData.addons;
+    let isDevDependencyInTwiddle = false;
+    Object.keys(twiddleAddons).forEach(item => {
+      if (!packageDevDep.hasOwnProperty(item)) {
+        isDevDependencyInTwiddle = true;
+        packageDevDep[item] = twiddleAddons[item];
+      }
+    });
+    // If there are dev dependencies present in twiddle, add them to the package json of the newly generated ember app.
+    if (isDevDependencyInTwiddle) {
+      fs.writeFileSync(`${emberAppDir}/package.json`, JSON.stringify(packageData));
```

Comment:
1) use path.join for windows compatibility

```suggestion
      fs.writeFileSync(`${path.join(emberAppDir, package)}.json`, JSON.stringify(packageData));
```

2) fs-extra makes this even better http://adilapapaya.com/docs/fs-extra/#writejsonfileobjectoptionscallback

```ts
import * as fs from 'fs-extra';

fs.writeJSONSync(`${path.join(emberAppDir, package)}.json`, packageData);
```